pynumaflow/batchmapper/servicer/async_servicer.py

```diff
@@ -0,0 +1,235 @@
+    messages in-flight at once.
+
+    This class is designed to enable different access patterns via inheritance. This can
+    be useful to build consistent data access ability across different applications.
+    In order to use this, the class being used must be provided to the BatchMapAsyncServer
+    for initialization and override present_data
+
+    Example Override:
+    class OtherImpl(BaseMapServicer):
+        async def present_data(self, datum_iterator: AsyncIterable[Datum])
+            -> AsyncIterable[batchmap_pb2.BatchMapResponse]:
+            ... # manage data here
+
+    """
+
+    def __init__(
+        self,
+        handler: MapBatchAsyncHandlerCallable,
+    ):
+        self._handler: MapBatchAsyncHandlerCallable = handler
+
+    async def IsReady(
+        self, request: _empty_pb2.Empty, context: NumaflowServicerContext
+    ) -> batchmap_pb2.ReadyResponse:
+        """
+        IsReady is the heartbeat endpoint for gRPC.
+        The pascal case function name comes from the proto batchmap_pb2_grpc.py file.
+        """
+        return batchmap_pb2.ReadyResponse(ready=True)
+
+    async def BatchMapFn(
+        self,
+        request_iterator: AsyncIterable[batchmap_pb2.BatchMapRequest],
+        context: NumaflowServicerContext,
+    ) -> AsyncIterable[batchmap_pb2.BatchMapResponse]:
+        """
+        Applies a batch function to a list of datum elements.
+        The pascal case function name comes from the proto batchmap_pb2_grpc.py file.
+        """
+        datum_iterator = datum_generator(request_iterator=request_iterator)
+
+        try:
+            # async for msg in self.__invoke_stream_batch(datum_iterator):
+            async for msg in self.present_data(datum_iterator):
+                yield msg
+        except Exception as err:
+            _LOGGER.critical("UDFError, re-raising the error", exc_info=True)
+            raise err
+
+    async def present_data(
+        self, datum_iterator: AsyncIterable[Datum]
+    ) -> AsyncIterable[batchmap_pb2.BatchMapResponse]:
+        try:
+            async for response_msgs in self._handler(datum_iterator):
+                # Implicitly ensure we have a list
+                yield batchmap_pb2.BatchMapResponse(
+                    results=[
+                        batchmap_pb2.BatchMapResponse.Result(
+                            keys=msg.keys, value=msg.value, tags=msg.tags
+                        )
+                        for msg in response_msgs.messages
+                    ],
+                    id=response_msgs.id,
+                )
+        except Exception as err:
+            err_msg = "UDSinkError: %r" % err
+            _LOGGER.critical(err_msg, exc_info=True)
+
+            async for _datum in datum_iterator:
+                yield batchmap_pb2.BatchMapResponse(
+                    batchmap_pb2.BatchMapResponse.Result.as_failure(_datum.id, err_msg)
+                )
+
+
+# ----
+class BatchMapUnaryServicer(BatchMapServicer):
```

Comment:
Similar to the other comments, let's leave this functionality for a later change. Right now lets just support a single interface for BatchMap